### PR TITLE
sign off / document first version of SQL schema.

### PR DESCRIPTION
**Pull request by bitonic**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql [hunk 1]
```diff
@@ -5,22 +5,38 @@
 --            As soon we released it, every schema change must be a separate file, so future migrations can work from any
 --            released Sandbox version.
 
+-- Stores the history of the ledger -- mostly transactions. This table is immutable in the sense that
+-- rows can never be modified, only added.
 CREATE TABLE ledger_entries
 (
+  -- Every entry is indexed by a monotonically growing integer. That is, new rows can only have a
+  -- ledger_offet which is greater than the larger ledger_offset in ledger_entries. However, note that there
+  -- might be gaps in the series formed by all the ledger_offsets in the table.
```
Comment (gaboraranyossy-da) [inline ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: let's wait with this statement until #752 lands

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql [hunk 2]
```diff
@@ -5,22 +5,38 @@
...
   ledger_offset         bigint primary key           not null,
-  -- one of 'transaction', 'rejection', or 'checkpoint' -- also see check_entry below
+  -- one of 'transaction', 'rejection', or 'checkpoint' -- also see check_entry below.
+  -- note that we _could_ store the different entries in different tables, but as of now we deem
```
Comment (gaboraranyossy-da) [inline ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: why the underscores?

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql [hunk 3]
```diff
@@ -5,22 +5,38 @@
...
+  -- more convient having a single table even if it imposes the constraint below, since
+  -- this table represents a single unified stream of events and partitioning it across table
```
Comment (gaboraranyossy-da) [inline ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: tables

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql [hunk 4]
```diff
@@ -40,32 +57,58 @@ CREATE TABLE disclosures (
...
   id             varchar primary key                                not null,
+  -- this is the transaction id that _originated_ the contract.
   transaction_id varchar references ledger_entries (transaction_id) not null,
+  -- this is the workflow id of the transaction above. note that this is
+  -- a denormalization -- we could simply look up in the transaction table.
+  -- we cache it here since we do not want to risk impacting performance
+  -- by looking it up in `ledger_entries`, however we should verify this
+  -- claim.
   workflow_id    varchar,
+  -- This triple is currently included in `contract`, since currently
```
Comment (gaboraranyossy-da) [inline ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: one more `currently` than needed

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql [hunk 5]
```diff
@@ -40,32 +57,58 @@ CREATE TABLE disclosures (
...
+  -- the identifier here separately.
   package_id     varchar                                            not null,
-  module_name    varchar                                            not null,
-  entity_name    varchar                                            not null,
-  create_offset  bigint references ledger_entries (ledger_offset)   not null,--TODO this is also denormalisation, as we could get this data from ledger_entries table too. We might not need this, this should be reviewed later.
+  -- using the QualifiedName#toString format
+  name           varchar                                            not null,
+  -- this is denormalized much like `transaction_id` -- see comment above.
+  create_offset  bigint references ledger_entries (ledger_offset)   not null,
+  -- this on the other hand _cannot_ be easily found out by looking into
```
Comment (gaboraranyossy-da) [inline ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: why the underscores?

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql (hunk 2 above)
Comment (bitonic) [reply ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: that's italics in plain text

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql (hunk 5 above)
Comment (bitonic) [reply ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: again, italics

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql (hunk 4 above)
Comment (bitonic) [reply ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: done

File: ledger/sandbox/src/main/resources/db/migration/V1__Init.sql (hunk 3 above)
Comment (bitonic) [reply ledger/sandbox/src/main/resources/db/migration/V1__Init.sql]: done

**Pull request merged**